### PR TITLE
Cleanup TextInfo tests

### PR DESCRIPTION
**Pull request by hughbe**
- Take advantage of xunit
- Cleanup multiple files for the same method into a single one (more
clean and easier to understand what's going on)

Contributes to #2336

Also fixes some nits that went unresolved in #6290

/cc @stephentoub

### CODE REVIEW

File: src/System.Globalization/tests/TextInfo/TextInfoCultureName.cs [hunk 1]
```diff
@@ -2,40 +2,22 @@
...
-            Assert.Equal(expectedLocaleName, new CultureInfo(localeName).TextInfo.CultureName);
-        }
-
-        // TestInvariantCultureName: Verify the invariant TextInfo
-        [Fact]
-        public void TestInvariantCultureName()
-        {
-            CultureInfo ci = CultureInfo.InvariantCulture;
-            string localeName = ci.Name;
-            Assert.Equal(ci.TextInfo.CultureName, localeName);
```
Comment (stephentoub) [inline src/System.Globalization/tests/TextInfo/TextInfoCultureName.cs]: What happened to this test?

**Comment by stephentoub** [comment]
Other than the one test that seems to have been deleted, LGTM.

**Comment by hughbe** [comment]
@stephentoub 
Do correct me if I'm wrong, but I think the test is here: https://github.com/hughbe/corefx/commit/990bf9907ae509fdb3bab68d8175e7b71b2e89d6#diff-b69db7f93df1499910d35d9c94df7fc0R12
IIRC, passing an empty string to the constructor of CultureInfo creates an InvariantCulture

**Comment by stephentoub** [comment]
> passing an empty string to the constructor of CultureInfo creates an InvariantCulture

It creates an invariant culture, but that's not exactly the same as verifying that the thing that CultureInfo.InvariantCulture returns also passes the test.

**Comment by hughbe** [comment]
Oh sure thing. I'll update this

**Comment by hughbe** [comment]
Ah right. I pushed a PR feedback commit adding back this test :)

**Comment by stephentoub** [comment]
Thanks :smile: LGTM

**Pull request merged**